### ISSUE TITLE
@JsonIgnore and @JsonProperty annotations don't work on interfaces

### ISSUE THREAD

**Issue by kumikumi** [owlike/genson#68]
Using @JsonIgnore annotation on a Java interface doesn't have an effect. Consider this example:

```java
public interface Person {
    String getFirstName();
    String getLastName();
    @JsonIgnore
    String getPassword();
    void setFirstName(String firstName);
    void setLastName(String lastName);
    void setPassword(String password);
}
```

```java
public class PersonImpl implements Person {
    private String firstName;
    private String lastName;
    private String password;
    public PersonImpl(String firstName, String lastName, String password) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.password = password;
    }
    @Override
    public String getFirstName() {
        return firstName;
    }
    @Override
    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }
    @Override
    public String getLastName() {
        return lastName;
    }
    @Override
    public void setLastName(String lastName) {
        this.lastName = lastName;
    }
    @Override
    @JsonIgnore
    public String getPassword() {
        return password;
    }
    @Override
    public void setPassword(String password) {
        this.password = password;
    }
}
```

The following code
```java
        Person p1 = new PersonImpl("John", "Doe", "hunter2");
        Genson genson = new GensonBuilder()
                .useIndentation(true)
                .useMethods(true)
                .create();
        System.out.println(genson.serialize(p1));
```

prints out as expected:

```
{
  "firstName":"John",
  "lastName":"Doe"
}
```

However, when I remove the @JsonIgnore annotation from the implementing class (and leave it just in the interface), the output is:
```
{
  "firstName":"John",
  "lastName":"Doe",
  "password":"hunter2"
}
```

For reference, using Jackson to serialize the interface with the following code
```
        Person p1 = new PersonImpl("John", "Doe", "hunter2");
        ObjectMapper mapper = new ObjectMapper();
        System.out.println(mapper.writeValueAsString(p1));
```

produces a json where the password is left out.

I also tried this briefly with @JsonProperty annotations and they didn't have an effect either when used with an interface.

Is this an intentional design decision or a bug with Genson? If this is intended behaviour, then the documentation may need to be improved.

**Comment by EugenCepoi** [comment]
It was partially by design. In the first versions I thought it would be better to keep annotations by method/field without merging them (so to ignore a getter and a setter you would have to annotate both of them). This defined also how inherited annotations would be handled = not merged.

But now, I think it would be better to merge them.

**Comment by EugenCepoi** [comment]
So I implemented a partial fix which is merging annotations between concrete implementations and abstract classes/interfaces, but it doesn't handle the JsonIgnore annotation (or any other annotation that would exclude the method from ser/de). This is mostly because in Genson the resolution of what method/field should be used is done first and then we merge things. However as the method is annotated with JsonIgnore we already excluded it and can't merge it.

In the future I will look into changing a bit this so your specific example can be addressed.
